hi-ml-histopathology/src/histopathology/configs/classification/BaseMIL.py

```diff
@@ -65,58 +64,35 @@ class BaseMIL(LightningContainer):
     batch_size: int = param.Integer(16, bounds=(1, None), doc="Number of slides to load per batch.")
     encoding_chunk_size: int = param.Integer(0, doc="If > 0 performs encoding in chunks, by loading"
                                                     "enconding_chunk_size tiles per chunk")
-    is_caching: bool = param.Boolean(False, doc="If True, cache the encoded tile features "
-                                     "(disables random subsampling of tiles). "
-                                     "If False (default), load the tiles without caching "
-                                     "(enables random subsampling of tiles).")
     # local_dataset (used as data module root_path) is declared in DatasetParams superclass
 
     def __init__(self, **kwargs: Any) -> None:
         super().__init__(**kwargs)
         self.best_checkpoint_filename = "checkpoint_max_val_auroc"
         self.best_checkpoint_filename_with_suffix = self.best_checkpoint_filename + ".ckpt"
-        self.checkpoint_folder_path = "outputs/checkpoints/"
-        self.callbacks = self.get_monitoring_callbacks(monitor="val/auroc", mode="max")
+        self.checkpoint_folder_path = f"{self.outputs_folder}/checkpoints/"
+        self.callbacks = self.get_callbacks()
 
     @property
     def cache_dir(self) -> Path:
-        return Path(
-            f"/tmp/innereye_cache1/{self.__class__.__name__}-{self.encoder_type}/"
-        )
+        return Path(f"himl_cache/{self.__class__.__name__}-{self.encoder_type}/")
 
     def setup(self) -> None:
         self.encoder = self.get_encoder()
         if not self.is_finetune:
             self.encoder.eval()
-        # Fine-tuning requires tiles to be loaded on-the-fly, hence, caching is disabled by default.
-        # When is_finetune and is_caching are both set, below lines should disable caching automatically.
-        if self.is_finetune:
-            self.is_caching = False
-        if not self.is_caching:
-            self.cache_mode = CacheMode.NONE
-            self.precache_location = CacheLocation.NONE
 
     def download_ssl_checkpoint(self, run_id: str) -> CheckpointDownloader:
         downloader = CheckpointDownloader(
             aml_workspace=get_workspace(),
             run_id=run_id,
-            checkpoint_filename="last.ckpt",
-            download_dir="outputs/",
+            checkpoint_filename=LAST_CHECKPOINT_FILE_NAME_WITH_SUFFIX,
+            download_dir=self.outputs_folder,
             remote_checkpoint_dir=Path(self.checkpoint_folder_path)
```

Comment:
[optional] With your other change below to `ModelCheckpoint.dirpath` in `get_callbacks()`, it looks like it would be safe to now just inline the value of `self.checkpoint_folder_path` here and get rid of the attribute. What do you think?

Reply:
What about get_path_to_best_checkpoint? Should I inline self.checkpoint_folder_path lines 173/179/184 as well?